core-java-modules/core-java-io-5/src/main/java/com/baeldung/usinggzipInputstream/Main.java

```diff
@@ -0,0 +1,41 @@
+package com.baeldung.usinggzipInputstream;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.zip.GZIPInputStream;
+
+public class Main {
+    static String filePath = Objects.requireNonNull(Main.class.getClassLoader().getResource("myFile.gz")).getFile();
+
+    public static void main(String[] args) throws IOException {
+
+        try {
+            readGZipFileWithFilter(filePath, lines -> lines.forEach(System.out::println));
+        } catch (FileNotFoundException e) {
+            System.err.println(e.getMessage());
```

Comment:
Let's avoid this pattern. Just outputting an error to `System.err` isn't helpful. Let's allow main to `throws` any exception it encounters here.